### PR TITLE
Added external usergroup entity

### PR DESCRIPTION
**Pull request by ntkathole**
```
>>> from nailgun import entities
>>> org = entities.Organization(name='test10').create()
>>> loc = entities.Location(name='test11').create()
>>> usergroup = entities.UserGroup(name="abc10").create()
>>> auth = entities.AuthSourceLDAP(onthefly_register=True, name='test4',server_type='free_ipa',port='389',account='uid=foreman,cn=example,dc=com',account_password='changeme',base_dn='cn=users,cn=example,dc=com', groups_base='cn=foobargroup,cn=example,dc=com',host='192.168.100.2', attr_login='uid',attr_lastname='givenName',attr_firstname='sn',attr_mail='mail', organization=[org],location=[loc]).create()
>>> entities.ExternalUserGroups(name='foobargroup',usergroups=usergroup,auth_source=auth).create()nailgun.entities.ExternalUserGroups(usergroups=nailgun.entities.UserGroup(admin=False, name='abc10', role=[], user=[], usergroup=[], id=3), name='foobargroup', auth_source=nailgun.entities.AuthSourceLDAP(id=16), id=4)
>>> entities.ExternalUserGroups(id=4, usergroups=usergroup).read()
nailgun.entities.ExternalUserGroups(usergroups=nailgun.entities.UserGroup(admin=False, name='abc10', role=[], user=[], usergroup=[], id=3), name='foobargroup', auth_source=nailgun.entities.AuthSourceLDAP(id=16), id=4)
>>> entities.ExternalUserGroups(id=4, usergroups=usergroup).delete()
{'id': 4, 'name': 'foobargroup', 'auth_source_id': 16, 'usergroup_id': 3}
```

### CODE REVIEW

**Comment by ntkathole** [comment]
@san7ket Updated!

File: nailgun/entities.py [hunk 1]
```diff
@@ -1116,6 +1116,93 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
             entity, attrs, ignore, params)
 
 
+class ExternalUserGroups(
```
Comment (oshtaier) [inline nailgun/entities.py]: all class names in nailgun has to be singular

**Comment by ntkathole** [comment]
Thanks @oshtaier updated!

**Comment by ntkathole** [comment]
```
>>> entities.ExternalUserGroup(name='foobargroup',usergroups=usergroup,auth_source=auth).create()
nailgun.entities.ExternalUserGroup(usergroups=nailgun.entities.UserGroup(admin=False, name='abc', role=[], user=[], usergroup=[], id=1), name='foobargroup', auth_source=nailgun.entities.AuthSourceLDAP(id=12), id=1)
>>> entities.ExternalUserGroup(id=1,usergroups=usergroup).refresh()
{'id': 1, 'name': 'foobargroup', 'auth_source_id': 12, 'usergroup_id': 1}
>>> entities.ExternalUserGroup(id=1,usergroups=usergroup).read()
nailgun.entities.ExternalUserGroup(usergroups=nailgun.entities.UserGroup(admin=False, name='abc', role=[], user=[], usergroup=[], id=1), name='foobargroup', auth_source=nailgun.entities.AuthSourceLDAP(id=12), id=1)
>>> entities.ExternalUserGroup(id=1,usergroups=usergroup).delete()
{'id': 1, 'name': 'foobargroup', 'auth_source_id': 12, 'usergroup_id': 1}
```

File: tests/test_entities.py [hunk 2]
```diff
@@ -442,6 +444,28 @@ def test_os_default_template(self):
                 self.cfg, id=2, operatingsystem=1).path()
         )
 
+    def test_refresh_externalusergroup(self):
```
Comment (oshtaier) [inline tests/test_entities.py]: hmm, report still showing that method as not covered:
image

Comment (abalakh) [reply tests/test_entities.py]: This test verifies `path()`, not `refresh()`, test name is misleading. `refresh()`, however, isn't covered at all.

File: nailgun/entities.py [hunk 3]
```diff
@@ -1116,6 +1116,93 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
             entity, attrs, ignore, params)
 
 
+class ExternalUserGroup(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin):
```
Comment (abalakh) [inline nailgun/entities.py]: Why no `EntityUpdateMixin`? PUT request is supported according to apidoc.

File: nailgun/entities.py [hunk 4]
```diff
@@ -1116,6 +1116,93 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
...
+    # Delete external usergroup
+    ExternalUserGroup(id=<id>, usergroups=usergroup).delete()
+    # Refresh external usergroup
+    ExternalUserGroup(id=<id>, usergroups=usergroup).refresh()
+    """
+    def __init__(self, server_config=None, **kwargs):
+        _check_for_value('usergroups', kwargs)
+        self._fields = {
+            'name': entity_fields.StringField(required=True),
+            'usergroups': entity_fields.OneToOneField(
```
Comment (abalakh) [inline nailgun/entities.py]: If it's one to one field (only one user group should be passed) - then pls rename it to `'usergroup'` accordingly.

File: tests/test_entities.py (hunk 2 above)
Comment (abalakh) [inline tests/test_entities.py]: pls rename to `test_externalusergroup` to avoid confusion.

**Comment by san7ket** [comment]
I suppose this would the ideal why , used should contribute henceforth in nailgun,
1. Docstring with example usage or entity request output
2. Unit test mandatory
3. Test output
@ntkathole +1

**Comment by ntkathole** [comment]
@abalakh @oshtaier all comments addressed. Thanks for review.

**Pull request merged**